components/button/w-button.vue

```diff
@@ -39,53 +39,58 @@ const props = defineProps({
-  [ccButton.pillSmallLoading]: props.pill && props.small && props.loading,
-  [ccButton.link]: props.link && !props.small,
-  [ccButton.linkSmall]: props.link && props.small,
-  [ccButton.linkAsButton]: !!props.href,
-  [ccButton.fullWidth]: props.fullWidth,
-  [ccButton.contentWidth]: !props.fullWidth,
-}));
+const primaryClasses = computed(() => [
+  !props.small && !props.quiet && !props.loading && ccButton.primary,
+  props.small && !props.quiet && !props.loading && ccButton.primarySmall,
+  props.small && props.quiet && !props.loading && ccButton.primarySmallQuiet,
+  props.small && props.loading && (props.quiet ? ccButton.primarySmallQuietLoading : ccButton.primarySmallLoading),
+  !props.small && props.quiet && !props.loading && ccButton.primaryQuiet,
+  !props.small && props.loading && (props.quiet ? ccButton.primaryQuietLoading : ccButton.primaryLoading),
+]);
+
+const secondaryClasses = computed(() => [
+  !props.small && !props.quiet && !props.loading && ccButton.secondary,
+  props.small && !props.quiet && !props.loading && ccButton.secondarySmall,
+  props.small && props.quiet && !props.loading && ccButton.secondarySmallQuiet,
+  props.small && props.loading && (props.quiet ? ccButton.secondarySmallQuietLoading : ccButton.secondarySmallLoading),
+  !props.small && props.quiet && !props.loading && ccButton.secondaryQuiet,
+  !props.small && props.loading && (props.quiet ? ccButton.secondaryQuietLoading : ccButton.secondaryLoading),
+]);
+
+const utilityClasses = computed(() => [
+  !props.small && !props.quiet && !props.loading && ccButton.utility,
+  props.small && !props.quiet && !props.loading && ccButton.utilitySmall,
+  props.small && props.quiet && !props.loading && ccButton.utilitySmallQuiet,
+  props.small && props.loading && (props.quiet ? ccButton.utilitySmallQuietLoading : ccButton.utilitySmallLoading),
+  !props.small && props.quiet && !props.loading && ccButton.utilityQuiet,
+  !props.small && props.loading && (props.quiet ? ccButton.utilityQuietLoading : ccButton.utilityLoading),
+]);
+
+const negativeClasses = computed(() => [
+  !props.small && !props.quiet && !props.loading && ccButton.negative,
+  props.small && !props.quiet && !props.loading && ccButton.negativeSmall,
+  props.small && props.quiet && !props.loading && ccButton.negativeSmallQuiet,
+  props.small && props.loading && (props.quiet ? ccButton.negativeSmallQuietLoading : ccButton.negativeSmallLoading),
+  !props.small && props.quiet && !props.loading && ccButton.negativeQuiet,
+  !props.small && props.loading && (props.quiet ? ccButton.negativeQuietLoading : ccButton.negativeLoading),
+]);
+
+const pillClasses = computed(() => [
+  !props.loading && (!props.small ? ccButton.pill : ccButton.pillSmall),
+  props.loading && (!props.small ? ccButton.pillLoading : ccButton.pillSmallLoading),
+]);
+
+const linkClasses = computed(() => [props.small ? ccButton.linkSmall : ccButton.link, !!props.href && ccButton.linkAsButton]);
```

Comment:
I forgot that there's also a check for `!!props.href`, and it is possible to use the button with `href` but without link, so the check in line 91 would not pass if only `href` was `true` 

Reply:
Updated in this commit: [600e483](https://github.com/warp-ds/vue/pull/187/commits/600e483c3dc19034a9823a653614e5139f60b2a1)